src/main/java/org/stellar/sdk/SorobanServer.java

```diff
@@ -0,0 +1,344 @@
+    } // TODO: Throw the appropriate exception.
+
+    SimulateTransactionResponse.SimulateHostFunctionResult simulateHostFunctionResult =
+        simulateTransactionResponse.getResults().get(0);
+
+    long classicFeeNum = transaction.getFee();
+    long minResourceFeeNum =
+        Optional.ofNullable(simulateTransactionResponse.getMinResourceFee()).orElse(0L);
+    long fee = classicFeeNum + minResourceFeeNum;
+    Operation operation = transaction.getOperations()[0];
+
+    if (operation instanceof InvokeHostFunctionOperation) {
+      Collection<SorobanAuthorizationEntry> originalEntries =
+          ((InvokeHostFunctionOperation) operation).getAuth();
+      List<SorobanAuthorizationEntry> newEntries = new ArrayList<>(originalEntries);
+      if (simulateHostFunctionResult.getAuth() != null) {
+        for (String auth : simulateHostFunctionResult.getAuth()) {
+          newEntries.add(sorobanAuthorizationEntryFromXdrBase64(auth));
+        }
+      }
+      operation =
+          InvokeHostFunctionOperation.builder()
+              .hostFunction(((InvokeHostFunctionOperation) operation).getHostFunction())
+              .sourceAccount(operation.getSourceAccount())
+              .auth(newEntries)
+              .build();
+    }
+
+    return new Transaction(
+        transaction.getAccountConverter(),
+        transaction.getSourceAccount(),
+        fee,
+        transaction.getSequenceNumber(),
+        new Operation[] {operation},
+        transaction.getMemo(),
+        transaction.getPreconditions(),
+        transaction.getSorobanData(),
+        transaction.getNetwork());
+  }
+
+  public SendTransactionResponse sendTransaction(Transaction transaction) throws IOException {
+    // TODO: In the future, it may be necessary to consider FeeBumpTransaction.
+    SendTransactionRequest params = new SendTransactionRequest(transaction.toEnvelopeXdrBase64());
+    return this.sendRequest(
+        "sendTransaction", params, new TypeToken<SorobanRpcResponse<SendTransactionResponse>>() {});
+  }
+
+  private <T, R> R sendRequest(
+      String method, @Nullable T params, TypeToken<SorobanRpcResponse<R>> responseType)
+      throws IOException {
+    String requestId = generateRequestId();
+    ResponseHandler<SorobanRpcResponse<R>> responseHandler = new ResponseHandler<>(responseType);
+    SorobanRpcRequest<T> sorobanRpcRequest = new SorobanRpcRequest<>(requestId, method, params);
+    MediaType mediaType = MediaType.parse("application/json");
+    RequestBody requestBody =
+        RequestBody.create(gson.toJson(sorobanRpcRequest).getBytes(), mediaType);
+
+    Request request = new Request.Builder().url(this.serverURI).post(requestBody).build();
+    try (Response response = this.httpClient.newCall(request).execute()) {
+      SorobanRpcResponse<R> sorobanRpcResponse = responseHandler.handleResponse(response);
+      if (sorobanRpcResponse.getError() != null) {
```

Comment:
thinking about this, not sure that a successful http response with json-rpc error in the response payload should be elevated to an exception, instead, just return the `sorobanRpcResponse` in all cases if you get it, as it has result and error attributes, that way clients only have to catch the IOException as a fatal type problem, otherwise they will inspect the response [result/error](https://github.com/stellar/java-stellar-sdk/pull/492/files#diff-db0499c80b2c87816a70ab620b64dea366e1b5d76c2fec3593c73db248d13a25R22) for further application meaning. I [commented similar on SorobanRpcErrorResponse](https://github.com/stellar/java-stellar-sdk/pull/492/files#r1276796480)

Reply:
Yes, personally I prefer returning the entire `SorobanRpcResponse`, but after reading the code of other libraries (including js-soroban-sdk), I found that they directly return RpcResponse.result.

Reply:
I have a question, is it necessary to expose the "id" field in jsonrpc to users? In other words, when making a jsonrpc request via http, is the "id" field almost useless?

Reply:
just in case, I'd wire it in for allowing it to be set by caller in sdk, but accepting null,etc so that a default hardcoded id can be applied, just so we don't get stuck.

Reply:
If that's the case, I will still stick to the current design, which is to throw a `SorobanRpcErrorResponse` exception if the response contains an error.

However, I will wrap the returned result, for example:
```java
@Value
@AllArgsConstructor
class WrappedResponse<T> {
  String id;
  T result;
}

public WrappedResponse<TransactionBuilderAccount> getAccount(String accountId)
```
Because interfaces like `getAccount` and `prepareTransaction` call RPC interfaces internally, but these interfaces do not directly return the data returned by Soroban-RPC to the user. This modification helps users obtain the ID.

Reply:
After careful consideration, I suggest sticking with the current design. If there is a need to expose the `id` in the future, we can add it without introducing any breaking changes..